src/Mailgun/Api/Routes.php

```diff
@@ -0,0 +1,157 @@
+    {
+        Assert::integer($limit);
+        Assert::integer($skip);
+        Assert::greaterThan($limit, 0);
+        Assert::greaterThanEq($skip, 0);
+
+        $params = [
+            'limit' => $limit,
+            'skip' => $skip,
+        ];
+
+        $response = $this->httpGet('/v3/routes', $params);
+
+        return $this->safeDeserialize($response, IndexResponse::class);
+    }
+
+    /**
+     * Returns a single Route object based on its ID.
+     *
+     * @param string $routeId Route ID returned by the Routes::index() method
+     *
+     * @return RouteDto
+     */
+    public function show($routeId)
+    {
+        Assert::stringNotEmpty($routeId);
+
+        $response = $this->httpGet(sprintf('/v3/routes/%s', $routeId));
+
+        return $this->safeDeserialize($response, ShowResponse::class);
+    }
+
+    /**
+     * Creates a new Route.
+     *
+     * @param string $expression  A filter expression like "match_recipient('.*@gmail.com')"
+     * @param array  $actions     Route action. This action is executed when the expression evaluates to True. Example: "forward('alice@example.com')"
+     * @param string $description An arbitrary string
+     * @param int    $priority    Integer: smaller number indicates higher priority. Higher priority routes are handled first. Defaults to 0.
+     *
+     * @return CreateResponse
+     */
+    public function create($expression, array $actions, $description, $priority = 0)
+    {
+        Assert::string($expression);
+        Assert::isArray($actions);
+        Assert::string($description);
+        Assert::integer($priority);
+
+        $params = [
+            'priority' => $priority,
+            'expression' => $expression,
+            'action' => $actions,
+            'description' => $description,
+        ];
+
+        $response = $this->httpPost('/v3/routes', $params);
+
+        return $this->safeDeserialize($response, CreateResponse::class);
+    }
+
+    /**
+     * Updates a given Route by ID. All parameters are optional.
+     * This API call only updates the specified fields leaving others unchanged.
+     *
+     * @param string      $routeId     Route ID returned by the Routes::index() method
+     * @param string|null $expression  A filter expression like "match_recipient('.*@gmail.com')"
+     * @param array|null  $actions     Route action. This action is executed when the expression evaluates to True. Example: "forward('alice@example.com')"
+     * @param string|null $description An arbitrary string
+     * @param int|null    $priority    Integer: smaller number indicates higher priority. Higher priority routes are handled first. Defaults to 0.
+     *
+     * @return UpdateResponse
+     */
+    public function update($routeId, $expression = null, array $actions = null, $description = null, $priority = null)
+    {
+        Assert::stringNotEmpty($routeId);
+
+        $params = [];
+
+        if (is_string($expression) && 0 < strlen($expression)) {
```

Comment:
I presume you mean `if (!empty($expression)) {` right? If I check just if the var is empty or not then I can pass `$expression = ['asdfg', 'qwerty']` and we won't get an error, when it's obvious that is wrong. This is why I am checking if we have an empty.

If you mean `if (is_string($expression) && !empty($expression))` then 👍